src/main/java/teammates/common/util/AdminLogQuery.java

```diff
@@ -0,0 +1,175 @@
+    private List<String> versionList;
+    
+    /**
+     * Default constructor.
+     */
+    public AdminLogQuery() {
+        query = LogQuery.Builder.withDefaults();
+        query.includeAppLogs(INCLUDE_APP_LOG);
+        query.batchSize(BATCH_SIZE);
+        query.minLogLevel(MIN_LOG_LEVEL);
+    }
+    
+    /**
+     * Sets values for query.
+     * If versionsToQuery is null or empty, default versions will be used instead.
+     * 
+     * @param versionsToQuery 
+     * @param startTime
+     * @param endTime
+     */
+    public AdminLogQuery(List<String> versionsToQuery, Long startTime, Long endTime) {
+        this();
+        setTimePeriodForQuery(startTime, endTime);
+        versionList = getVersionIdsForQuery(versionsToQuery);
+        query.majorVersionIds(versionList);
+    }
+    
+    /**
+     * Gets query to retrieve logs.
+     */
+    public LogQuery getQuery() {
+        return query;
+    }
+    
+    /**
+     * Sets time period to search for query.
+     * @param startTime
+     * @param endTime
+     */
+    private void setTimePeriodForQuery(Long startTime, Long endTime) {
+        if (startTime != null) {
+            query.startTimeMillis(startTime);
+        }
+        if (endTime != null) {
+            query.endTimeMillis(endTime);
+            setEndTime(endTime);
+        }
+    }
+    
+    /**
+     * Sets time period to query logs then moves endTime back by the same amount.
+     * @param timeInMillis milliseconds before endTime
+     */
+    public void setQueryWindowBackward(long timeInMillis) {
+        if (getEndTime() == null) {
+            setEndTime(TimeHelper.now(0.0).getTimeInMillis());
+        }
+        Long startTime = getEndTime() - timeInMillis;
+        setTimePeriodForQuery(startTime, getEndTime());
+        endTime = startTime - 1;
+    }
+    
+    /**
+     * Sets end time of the query.
+     */
+    private void setEndTime(Long endTimeParam) {
+        endTime = endTimeParam;
+    }
+    
+    /**
+     * Gets end time of the query.
+     */
+    public Long getEndTime() {
+        return endTime;
+    }
+    
+    /**
+     * Gets versions used in query.
+     */
+    public List<String> getVersionsToQuery() {
+        return versionList;
+    }
+    
+    /**
+     * Selects versions for query. If versions are not specified, it will return 
+     * default versions used for query.
+     */
+    private List<String> getVersionIdsForQuery(List<String> versions) {
+        boolean isVersionSpecifiedInRequest = (versions != null && !versions.isEmpty());
+        if (isVersionSpecifiedInRequest) {
+            return versions;
+        }
+        return getDefaultVersionIdsForQuery();
+    }
+    
+    /**
+     * Gets a list of versions, including the current version and 5 preceding versions (if available).
+     * @return a list of default versions for query.
+     */
+    private List<String> getDefaultVersionIdsForQuery() {
+        GaeVersionApi adminApi = new GaeVersionApi();
```

Comment:
This dependency doesn't feel right. This class has no business knowing the `GaeVersionApi` class. The call site (i.e. probably the `Action` class) should get the relevant stuff from the api and pass to this class. 
In fact, this whole method feels like it should be in `GaeVersionApi` class.